pos_accounting/src/storage/mod.rs

```diff
@@ -34,7 +32,7 @@ pub trait PoSAccountingStorageWrite: PoSAccountingStorageRead {
 
     fn del_pool_balance(&mut self, pool_id: H256) -> Result<(), Error>;
 
-    fn set_pool_data(&mut self, pool_id: H256, pool_data: &PoolData) -> Result<(), Error>;
+    fn set_pool_data(&mut self, pool_id: H256, pool_data: PoolData) -> Result<(), Error>;
```

Comment:
In theory, setters should take things by value, not by reference. But in practice, for a database, they should be by reference. The reason is that setters for databases don't really consume values. I know you're dealing with maps now for the in-memory storage, but this isn't really how this works in practice for databases. Imagine you want to store PoolData in the database with lmdb or leveldb. Basically, the first thing you'll do is that you'll have to serialize PoolData, which only uses a reference of PoolData, causing an extra unnecessary copy to be destroyed.

If you wish to make this "perfect", you can use the std::borrow::Cow type, but then the problem there is that you can't enforce one way or the other. So users can still pass things by ref when it's in memory.

Or you could support both interfaces, which again has the same issue.

So, the eventually, you'll remember that the only purpose of the that map is testing, so cloning from the ref into the map isn't such a bad idea. It's the right way to do things since serialization is equivalent to cloning, in a way.